src/main/scala/mesosphere/marathon/core/storage/repository/impl/DeploymentRepositoryImpl.scala

```diff
@@ -57,32 +65,47 @@ object StoredPlan {
   }
 
   def apply(proto: Protos.DeploymentPlanDefinition): StoredPlan = {
-    val version = if (proto.hasTimestamp()) {
-      OffsetDateTime.parse(proto.getTimestamp, DateFormat)
+    val version = if (proto.hasTimestamp) {
+      Try(OffsetDateTime.parse(proto.getTimestamp, DateFormat)).getOrElse(OffsetDateTime.MIN)
     } else {
       OffsetDateTime.MIN
     }
     StoredPlan(
       proto.getId,
-      OffsetDateTime.parse(proto.getOriginalRootVersion, DateFormat),
-      OffsetDateTime.parse(proto.getTargetRootVersion, DateFormat),
+      Try(OffsetDateTime.parse(proto.getOriginalRootVersion, DateFormat)).getOrElse(OffsetDateTime.MIN),
+      Try(OffsetDateTime.parse(proto.getTargetRootVersion, DateFormat)).getOrElse(OffsetDateTime.MIN),
       version)
   }
 }
 
+// TODO: We should probably cache the plans we resolve...
+// TODO: Add more docs, especially about GC.
 class DeploymentRepositoryImpl[K, C, S](
     persistenceStore: PersistenceStore[K, C, S],
-    groupRepository: GroupRepository)(implicit
+    groupRepository: StoredGroupRepositoryImpl[K, C, S],
+    appRepository: AppRepositoryImpl[K, C, S],
+    maxVersions: Int)(implicit
   ir: IdResolver[String, StoredPlan, C, K],
     marshaller: Marshaller[StoredPlan, S],
     unmarshaller: Unmarshaller[S, StoredPlan],
-    ctx: ExecutionContext) extends DeploymentRepository {
+    ctx: ExecutionContext,
+    actorRefFactory: ActorRefFactory,
+    mat: Materializer) extends DeploymentRepository {
+
+  private val gcActor = actorRefFactory.actorOf(
```

Comment:
Only context information: in metronome we tried to use factories for props, that are defined in the `Module` class, if the actor has dependencies.
(E.g.: https://github.com/dcos/metronome/blob/master/jobs/src/main/scala/dcos/metronome/jobrun/JobRunModule.scala#L30)

All dependencies can be resolved in the Module and does not need to be defined and delegated through the instantiating class.